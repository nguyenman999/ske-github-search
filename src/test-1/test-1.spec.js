import React from 'react'
import { shallow } from 'enzyme'
import Test1 from './test-1'

describe('test-1-view', () => {
  let props

  beforeEach(() => {
    props = {
      q: 'search string',
      page: 1,
      loading: false,
      loadMore: false,
      list: [],
      actions: {
        onSearchInputChange: jest.fn(),
        searchUsers: jest.fn(),
        loadMoreUsers: jest.fn(),
      },
    }
  })

  it('render default', () => {
    const wrapper = shallow(<Test1 {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('render loadmore', () => {
    props = {
      ...props,
      loadMore: true,
      list: [
        {
          avatarUrl: 'avatarUrl',
          score: 123,
          login: 'login',
          type: 'type',
        },
      ],
    }
    const wrapper = shallow(<Test1 {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('onInputChange should dispatch action onSearchInputChange', () => {
    const wrapper = shallow(<Test1 {...props} />)
    const instance = wrapper.instance()
    instance.onInputChange({ target: 'target' })
    expect(props.actions.onSearchInputChange).toHaveBeenCalled()
  })

  it('onInputSearch should dispatch action searchUsers', () => {
    const wrapper = shallow(<Test1 {...props} />)
    const instance = wrapper.instance()
    instance.onInputSearch()
    expect(props.actions.searchUsers).toHaveBeenCalled()
  })

  it('onLoadMore should dispatch action loadMoreUsers', () => {
    const wrapper = shallow(<Test1 {...props} />)
    const instance = wrapper.instance()
    instance.onLoadMore()
    expect(props.actions.loadMoreUsers).toHaveBeenCalled()
  })

  it('renderItem', () => {
    const wrapper = shallow(<Test1 {...props} />)
    const instance = wrapper.instance()
    const result = instance.renderItem({
      avatarUrl: 'avatarUrl',
      score: 123,
      login: 'login',
      type: 'type',
    })
    expect(shallow(result)).toMatchSnapshot()
  })
})
