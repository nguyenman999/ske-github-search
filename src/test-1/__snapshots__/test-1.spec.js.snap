// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-1-view render default 1`] = `
<div
  className="container"
>
  <List
    bordered={false}
    dataSource={Array []}
    header={
      <UserListHeader
        onChange={[Function]}
        onSearch={[Function]}
        title="Github Search User"
      />
    }
    loadMore={null}
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
</div>
`;

exports[`test-1-view render loadmore 1`] = `
<div
  className="container"
>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "avatarUrl": "avatarUrl",
          "login": "login",
          "score": 123,
          "type": "type",
        },
      ]
    }
    header={
      <UserListHeader
        onChange={[Function]}
        onSearch={[Function]}
        title="Github Search User"
      />
    }
    loadMore={
      <div
        style={
          Object {
            "height": 32,
            "lineHeight": "32px",
            "marginTop": 12,
            "textAlign": "center",
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
          loading more
        </Button>
      </div>
    }
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
</div>
`;

exports[`test-1-view renderItem 1`] = `
<Item>
  <Skeleton
    active={true}
    avatar={true}
    loading={false}
    paragraph={true}
    title={false}
  >
    <Meta
      avatar={
        <Avatar
          shape="circle"
          size="default"
          src="avatarUrl"
        />
      }
      description="type"
      title="login"
    />
    <div>
      123
    </div>
  </Skeleton>
</Item>
`;
